src/daria/image/image.py

```diff
@@ -49,38 +50,68 @@ class Image:
-        # It should include the format of timestamps included in the exifdata
+        img: Union[np.ndarray, str],
+        metadata: Optional[dict] = None,
+        curvature_correction: Optional[da.CurvatureCorrection] = None,
+        color_correction: Optional[da.ColorCorrection] = None,
+        **kwargs,
     ) -> None:
         """Constructor of Image object.
 
         The input image can either be a path to the image file or a numpy array with
         conventional matrix indexing.
 
         Arguments:
-            img (np.ndarray): image array with matrix indexing
-            origo (np.ndarray): physical coordinates of the lower left corner
-            width (float): physical width of the image
-            height (float): physical height of the image
-            depth (float): physical depth of the image, only relevant for 3d images
-            dim (int): dimension of image, could be 2, 3, 4 (incl. the possibility for time)
-            read_metadata_from_file (bool): controlling whether meta data is read from file
-            metadata_path (str): path to metadata; should be changed if metadata is not placed
-                in the "images/metadata"-folder and taking the same path as the image. If
-                image is not passed as a path one also needs to pass it as a path.
+            img (Union[np.ndarray, str]): image array with matrix indexing
+            metadata (dict, Optional): metadata dictionary, default is None.
+            curvature_correction (daria.CurvatureCorrection, Optional):
+                Curvature correction object. Default is none, but should be included
+                if the image is to be curvature corrected at initialization
+            color_correction (daria.ColorCorrection, Optional): Color correction object.
+                Default is none, but should be included if the image is to be color
+                corrected at initialization.
+            kwargs (Optional arguments)
+                metadata_source (str): Path to a metadata json-file that provides
+                    metadata such as physical width, height and origo of image
+                    as well as colorspace
+                origo (np.ndarray): physical coordinates of the lower left corner
+                width (float): physical width of the image
+                height (float): physical height of the image
+                color_space (str): the colorspace of the image. So far only BGR
+                    and RGB are "valid", but more should be added at a later time.
         """
 
+        # Read metadata
+        no_colorspace_given = False
+        if metadata is not None:
+            self.metadata = metadata
+            self.width: float = self.metadata["width"]
+            self.height: float = self.metadata["height"]
+            self.origo: list[float] = self.metadata["origo"]
+            self.colorspace: str = self.metadata["color_space"]
```

Comment:
One could ask, why having the same information stored twice, as metadata is part of `Image`, storing `width`, `height`, `origo`, and `color_space`, at the same time, these are stored as attributes of `Image`. Furthermore, they are always aimed at being synchronized. Why then not only using `metadata`. Then one could remove the marked lines, and similarly lines 98-101, just below. Of course, this would require some changes in the way `metadata` is updated, but this would not be difficult either.

Reply:
I agree. I was just afraid that it would make it more inconvenient to extract the attributes from ouside of the object. However, I don't think we'll ever need to, and it is not really much more work...

Reply:
> View / edit / reply to [this conversation](https://app.reviewnb.com/pmgbergen/DarIA/pull/57/discussion/) on ReviewNB
> 
> _jwboth commented on 2022-09-24T19:59:39Z_ ----------------------------------------------------------------
> 
> Line #1.    grid_image = baseline_image.add_grid(dx = 10, dy = 10, color= (150,0, 100))
> Is it actually clear what units `dx` and `dy` are using? We should either define a convention (like using mm, cm, m, pixels) or we even have to include some indicator for units. We may need to provide capapbility for the conversion then as well. Adding such a possibility would possibly also allow better understanding of the input arguments.
> 
> While thinking about the arguments of `add_grid`. Would it maybe be better to also allow for `str` arguments for color and define some standard colors, which then in a utils method are converted to the correct RGB values?
> 
> 
> _EStorvik commented on 2022-09-25T05:44:59Z_ ----------------------------------------------------------------
> 
> I guess they are based on the "width" and "height" of the image, and are somehow automatically in the same units. Regarding the str I agree.

Reply:
hmm, OK I see your point. we can discuss tomorrow. Maybe it is then safer to keep it?